pgjdbc/src/main/java/org/postgresql/core/PGStream.java

```diff
@@ -643,6 +647,50 @@ public void setMaxResultBuffer(String value) throws PSQLException {
     maxResultBuffer = PGPropertyMaxResultBufferParser.parseProperty(value);
   }
 
+  /**
+   * Method to get MaxResultBuffer from PGStream.
+   *
+   * @return size of MaxResultBuffer
+   */
+  public long getMaxResultBuffer() {
+    return maxResultBuffer;
+  }
+
+  /**
+   * The idea behind this method is to keep in maxRowSize the size of biggest read data row. As
+   * there may be many data rows send after each other for a query, then value in maxRowSize would
+   * contain value noticed so far, because next data rows and their sizes are not read for that
+   * moment. We want it increasing, because the size of the biggest among data rows will be used
+   * during computing new adaptive fetch size for the query.
+   *
+   * @param rowSize new value to be set as maxRowSize
+   */
+  public void setMaxRowSize(int rowSize) {
+    if (maxRowSize == -1) {
```

Comment:
if rowsize is always positive the check for -1 is not necessary

Reply:
That's true, this check is not necessary, just wanted to show that if there is no maxRowSize then given rowSize should be set as maxRowSize without checking.